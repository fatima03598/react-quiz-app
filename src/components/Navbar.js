
import React from 'react';
import {
  Navbar,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';

const NavigationBar = (props) => {
 

  return (
    <div>
      <Navbar color="light" light expand="md">
        <NavbarBrand href="/">QuizName</NavbarBrand>
          <Nav className="mr-auto" >
            <NavItem>
              <NavLink href="/">New Game</NavLink>
            </NavItem>
          </Nav>
      </Navbar>
    </div>
  );
}

export default NavigationBar;

