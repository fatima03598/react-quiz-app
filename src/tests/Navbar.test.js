import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '../components/Navbar';
import { shallow, mount } from 'enzyme';
import { Link, Route, Switch, BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
import NavigationBar from '../components/Navbar';

describe('Navbar', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<NavigationBar />);
    });
  
    it('matches the snapshot', () => {
    const tree = renderer.create(<Navbar />).toJSON();
      expect(tree).toMatchSnapshot();
    });

  
    it('includes link to homepage', () => {                                       
      expect(wrapper.find("[href='/']")).toBeTruthy();
     });

     it('includes link to New Game', () => {                                       
      expect(wrapper.find("[href='/newgame']")).toBeTruthy();
     });

     it('includes link to Past Scores', () => {                                       
      expect(wrapper.find("[href='/pastscores']")).toBeTruthy();
     });
  
    
  });


