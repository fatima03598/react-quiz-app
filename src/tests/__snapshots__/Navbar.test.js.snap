// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches the snapshot 1`] = `
<div
  className="navbar-container"
>
  <nav>
    <a
      className="nav-link"
      href="/"
    >
       FaRRoaH Quiz
    </a>
    <a
      className="nav-link"
      href="/"
    >
      New Game
    </a>
  </nav>
</div>
`;
